src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/OrderStateController.php

```diff
@@ -377,6 +437,98 @@ public function toggleSendEmailAction($orderStateId)
         return $this->redirectToRoute('admin_order_states');
     }
 
+    /**
+     * Deletes order state
+     *
+     * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))", redirectRoute="admin_order_states")
+     *
+     * @param int $orderStateId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction(Request $request, int $orderStateId): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteOrderStateCommand($orderStateId));
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion', 'Admin.Notifications.Success')
+            );
+        } catch (OrderStateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $request->query->has('redirectUrl') ?
+            $this->redirect($request->query->get('redirectUrl')) :
+            $this->redirectToRoute('admin_order_states');
+    }
+
+    /**
+     * Delete order states in bulk action.
+     *
+     * @AdminSecurity(
+     *     "is_granted('delete', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_order_states",
+     *     message="You do not have permission to delete this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function deleteBulkAction(Request $request): RedirectResponse
+    {
+        $orderStateIds = $this->getBulkOrderStatesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkDeleteOrderStateCommand($orderStateIds));
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion', 'Admin.Notifications.Success')
+            );
+        } catch (OrderStateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_order_states');
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function getBulkOrderStatesFromRequest(Request $request): array
+    {
+        $orderStateIds = $request->request->get('order_states_order_states_bulk');
+
+        if (!is_array($orderStateIds)) {
+            return [];
+        }
+
+        return array_map(static function (string $orderStateId) {
```

Comment:
Why a static function instead of a closure?

Reply:
Fixed